### DISCUSSION TITLE
Error on initial setup with Slingbox

### DISCUSSION THREAD

**Discussion by donscreen** [GerryDazoo/Slinger#87] [Q&A]
I am running the exe server file on Windows 10.   It does see both of my slingboxes.  When I use VLC to view the network stream, I am getting this error.  I am trying to connect to the network stream at http://192.168.1.173:8082/slingbox.  I have a 500 and an M2 slingbox.  I get the same error when trying to stream from either box.
Windows firewall is turned off.  I have port forwarded 192.168.1.173 to port 8082 and the slingbox at ip 192.168.1.108 to port 5316

![192109835-7ca6f186-d2d9-449b-b349-67341c2f4853](https://user-images.githubusercontent.com/114305703/192110160-08438528-8cdd-418e-b152-e94710b2251f.png)

My config.ini file....passwords have been blanked but are there and correct.
[SLINGBOX]
sbtype="350/500/M1"
;sbtype="Solo/Pro/ProHD"
password=
; uncomment the next lines ipaddress and port with your local network info
; if you don't want the server to automatically find your 
; slingbox on the local network. Needed if server and slingbox are
; not on the same LAN segment. ALso removes requirement for the netifaces module  
ipaddress=192.168.1.108
port=5316
; Repace with your finderid
;finderid=BAD95DAED609364C8204F2BC5FBDC363

;name=something

; Following are the default values
; Default resolution when server starts. You can change this and reconnect without having to restart the server
; valid range = 1..15
;Resolution=12

; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
;FrameRate=30

;Min 50 Max 8000
;VideoBandwidth=2000

; Min 0 (auto) max 63
;VideoSmoothness=63

; Send Iframe every n seconds Max 30
;IframeRate=5

; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
;AudioBitRate=32

; Video Source  0, 1, 2, 3 depending on your hardware corrosponds to one of
;Composite, Component, HDMI or Tuner.
;ProHD 0=Tuner, ???
;Solo, M1 0=composite 1=Svideo 2=component
;500  0=composite, 1=component, 2=HDMI
; If you don't specify it will default to the last selected input
; So if you only have one video source connected you don't need to 
; mess with this. This number "might" be the same as the [REMOTE] code= entry but I can't say for sure. Please ket me know if you find a counter example
;VideoSource = 1

[SERVER]
; local port number for the server to listen on for connections
port=8082
maxstreams=10
enableremote=yes

[REMOTE]
; see release notes for changing this if your remote doesn't work
;code=2
; path to an external remote control definitions file, if the defaults
; aren't working for you or you want to change the format of the web page.
; use the supplied remote.txt as a starting point.
;include=remote.txt

**Comment by GerryDazoo** [comment]
I suspect, you're not connecting to the slingbox you think you are. I'm pretty sure when the code says your password is wrong, your password is wrong. Start a viewing session with the sling software and very /settings/Network Info

**Reply by donscreen** [reply]
I have always used the same password for my slingbox.  I tried admin just in case.....still same error.
![image](https://user-images.githubusercontent.com/114305703/192111547-20995536-f1b7-4a10-8585-48ff399f8cf1.png)

**Reply by GerryDazoo** [reply]
Did you get your admin password using the instructions in the readme?  You did say you had two boxes. How do you know for sure you have the right password for this box. I have no doubt you're connecting to 192.168.1.108:5316.

**Reply by donscreen** [reply]
I went to the link in the readme file.  I even logged out of my slingbox account.   I logged back in with the same admin password and was able to see both of my slingboxes in the directory.   Same password gives me access to either box.

**Reply by donscreen** [reply]
Thank you!!!  I appreciate your patience with me.   The readme link to retrieve  the admin password for each box did work.  I had never seen those particular passwords before.  Just my account admin password.   Thank you so much for all of your help and creating this software!

**Comment by eskwadrat** [comment]
Here is what happens with my M1 configured for Slinger:

Version : 3.07 Running on Windows-10-10.0.19042-SP0 pid= 8268
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running:  10 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('192.168.0.241', 5301)
('192.168.0.241', 5301) OK
Using slingbox at  ('192.168.0.241', 5301)
Streamer:  PLM1UPC Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from remote.txt
Error reading remote definition file, using defaults
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses (0.0.0.0)
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://127.0.0.1:8081
 * Running on http://192.168.0.220:8081 (Press CTRL+C to quit)

When I try stream using VLC (on PC in same LAN), here is what happens:
* on server console:

192.168.0.220 - - [29/Sep/2022 12:56:49] "GET / HTTP/1.1" 404 -
192.168.0.220 - - [29/Sep/2022 12:56:49] "GET / HTTP/1.0" 404 -
192.168.0.220 - - [29/Sep/2022 12:58:11] "GET / HTTP/1.1" 404 -
192.168.0.220 - - [29/Sep/2022 12:58:11] "GET / HTTP/1.0" 404 -

the VLC error windows lists:
Your input can't be opened:
VLC is unable to open the MRL 'http://192.168.0.220:8081'. Check the log for details.
Your input can't be opened:
VLC is unable to open the MRL 'http://192.168.0.220:8081'. Check the log for details.

Suggestions? 
also, I can't find my M1 finderID, where to look for it? I am using IP in the .ini file.

**Reply by GerryDazoo** [reply]
You don't need the finderid unless you want to access your server remotely.
Where did you get the password for your M1?

**Reply by eskwadrat** [reply]
It is same password I setup 15 yrs ago which allows me to login to my Slingbox account to see all my slingboxes and reconfigure them. Same password to stream from each one of those on my PCs or via FireTV stick Slingbox app. It is admin password, I have never setup any other password than this. I followed the link in your README file and logged in successfully on the Slingmedia website. 

Are you suggesting there might be separate password for my M1?  Thank you for being patient with me.

**Reply by sp222222** [reply]
you need to get your password from here before end of the month
https://newwatchsecure.slingbox.com/watch/slingAccounts/account_boxes_js

**Comment by GerryDazoo** [comment]
Streamer Running: 10 config.ini 8080

You’re using the wrong URL ALL the examples say ;

http://192.168.0.220:8080/slingbox  to stream video

http://192.168.0.220:8080/Remote   for the remote control

**Comment by eskwadrat** [comment]
GerryDazoo, thanks!
This is what I get next, says password is incorrect but I triple checked it is correct, it is the password I login to my Slingplayer app and FireStick SlingPlayer app version. Can't be wrong. What am I missing? 

09/29/2022, 13:18:12   Streaming connection from ('192.168.0.220', 51071)

Slinginfo  "350/500/M1" password 12 30 192.168.0.241 5301 3072
PLM1UPC Starting Stream for  192.168.0.220:51071
Connecting... ('192.168.0.241', 5301) Control
cmd: 0x67 err: 0x2 8
PLM1UPC Error Starting Session. Check your admin password in config.ini file!
PLM1UPC ERROR: Slingbox session startup failed.
Streamer:  PLM1UPC Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Comment by GerryDazoo** [comment]
Sorry, Steve, it CAN be wrong. The password for the slingbox is NOT the same as your slingplayer account. Please read the read me on how to recover your slingbox password. You obviously have not done as the readme suggests, otherwise you would have the slingbox admin password and finderid.
https://newwatchsecure.slingbox.com/watch/slingAccounts/account_boxes_js

**Comment by eskwadrat** [comment]
Thanks, I read readme file couple times and used the link above. With Slingbox disconnected it takes me to Slingbox website, hence my confusion. With M1 in network and active I got all the passwords, super cool. Yes, they are completely different from my account password and to be honest I wasn't even aware they exist. For all my 4 Slingboxes I had in past 15yrs or so. 

Anyhow, I entered correct password and now the server seem to be moving forward with connection. When I try streaming via VLC, I see these messages and after about 30 seconds the server closes (crashes?) and I get nothing on the VLC. Here is the server console:

Version : 3.07 Running on Windows-10-10.0.19042-SP0 pid= 876
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running:  10 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('192.168.0.241', 5301)
('192.168.0.241', 5301) OK
Using slingbox at  ('192.168.0.241', 5301)
Streamer:  PLM1UPC Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses (0.0.0.0)
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://127.0.0.1:8081
 * Running on http://192.168.0.220:8081 (Press CTRL+C to quit)
09/30/2022, 09:09:06   Streaming connection from ('192.168.0.220', 50286)

Slinginfo  "350/500/M1" 14i7p88p0g194gy 12 30 192.168.0.241 5301 3072
PLM1UPC Starting Stream for  192.168.0.220:50286
Connecting... ('192.168.0.241', 5301) Control
PLM1UPC Selecting Video Source 2

<here after about 20-30 sec the server closes window>

**Reply by eskwadrat** [reply]
update: I tried few times and this time, the server do not close/crash but I still get nothing on VLC , the console says (see last line). Note: I always thought the Slingox stream is encrypted, so how possible VLC can play it back without decryption? Perhaps that is why I get no picture on the VLC. ?

Version : 3.07 Running on Windows-10-10.0.19042-SP0 pid= 4480
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running:  10 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('192.168.0.241', 5301)
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses (0.0.0.0)
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://127.0.0.1:8081
 * Running on http://192.168.0.220:8081 (Press CTRL+C to quit)
Error connecting to  ('192.168.0.241', 5301)
No valid slingbox ip info found in config.ini
Finding Slingbox on local network. My IP Info =  fe80::f99a:2fc1:8f94:986a%8
Finding Slingbox on local network. My IP Info =  192.168.0.220
Finding Slingbox on local network. My IP Info =  ::1
Finding Slingbox on local network. My IP Info =  127.0.0.1
Exception in thread Thread-2 (streamer):
Traceback (most recent call last):
  File "threading.py", line 1016, in _bootstrap_inner
  File "threading.py", line 953, in run
  File "slingbox_server.py", line 539, in streamer
    for port in stati.keys:
TypeError: 'builtin_function_or_method' object is not iterable
09/30/2022, 09:15:54   RemoteControl connection from ('192.168.0.220', 50375)

Starting remote control stream handler for  ('192.168.0.220', 50375) to port 8081
Remote Control Connected
127.0.0.1 - - [30/Sep/2022 09:15:54] "GET /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.0.220', 50375)
09/30/2022, 09:17:04   RemoteControl connection from ('192.168.0.220', 50382)

Starting remote control stream handler for  ('192.168.0.220', 50382) to port 8081
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('4', '12 : round'), ('Digits', '')])
127.0.0.1 - - [30/Sep/2022 09:17:04] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.0.220', 50382)
Hacker Alert. Invalid Request from  ('192.168.0.220', 50388)
Hacker Alert. Invalid Request from  ('192.168.0.220', 50389)
192.168.0.220 - - [30/Sep/2022 09:17:50] "GET /slingbox HTTP/1.1" 404 -
192.168.0.220 - - [30/Sep/2022 09:17:50] "GET /slingbox HTTP/1.0" 404 -
09/30/2022, 09:17:56   Streaming connection from ('192.168.0.220', 50392)

**Reply by GerryDazoo** [reply]
So many things here.
** This is good **
Checking for slingbox at ('192.168.0.241', 5301)
('192.168.0.241', 5301) OK
*************************
** This is bad ****
Error connecting to ('192.168.0.241', 5301)
No valid slingbox ip info found in config.ini
It means your slingbox is dropping off the network and the server can't find it.
****************************************************
This the last time I'll say this. You need to start a video stream to get the Remote to work.
In your case, your last logs, the stream isn't starting because the server can't find a slingbox to connect to.
The other main reason for a blank screen is you've selected the wrong VideoSource.
One more time,,, Open a command prompt and run the server manually. Don't just click on the exe. That way the window will stay open so you can se what's happening.
No the video stream from the slingbox is not encrypted.

**Reply by eskwadrat** [reply]
OK, I did like you said, via cmd prompt. Looks like there is some issue with 18 bytes buffer your server cannot
"PLM1UPC Badness starting slingbox session  unpack requires a buffer of 18 bytes None"
Looks to me like a script issue. 

Here is a full console feedback (no remote is used this time). 

Microsoft Windows [Version 10.0.19042.1466]
(c) Microsoft Corporation. All rights reserved.

Version : 3.07 Running on Windows-10-10.0.19042-SP0 pid= 10112
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running:  10 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('192.168.0.241', 5301)
('192.168.0.241', 5301) OK
Using slingbox at  ('192.168.0.241', 5301)
Streamer:  PLM1UPC Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses (0.0.0.0)
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://127.0.0.1:8081
 * Running on http://192.168.0.220:8081 (Press CTRL+C to quit)
09/30/2022, 17:25:15   Streaming connection from ('192.168.0.220', 59069)

Slinginfo  "350/500/M1" 14i7p88p0g194gy 12 30 192.168.0.241 5301 3072
PLM1UPC Starting Stream for  192.168.0.220:59069
Connecting... ('192.168.0.241', 5301) Control
PLM1UPC Selecting Video Source 2
Traceback (most recent call last):
  File "slingbox_server.py", line 678, in streamer
    s_ctl, stream  = start_slingbox_session(streams)
  File "slingbox_server.py", line 427, in start_slingbox_session
    sling_cmd(0x85, pack('4h', source, 0, 0, 0 ))
  File "slingbox_server.py", line 348, in sling_cmd
    sid, stat, dlen = unpack("2x H 8x H 2x H", response[0:18] ) # "x2 v x8 v x2 v", $hbuf);
struct.error: unpack requires a buffer of 18 bytes
PLM1UPC Badness starting slingbox session  unpack requires a buffer of 18 bytes None
ERROR: The process with PID 1608 (child process of PID 4496) could not be terminated.
Reason: The process cannot terminate itself.
SUCCESS: The process with PID 4496 (child process of PID 10112) has been terminated.
SUCCESS: The process with PID 10112 (child process of PID 9052) has been terminated.

C:\Slinger\Slinger-main\Slingbox>

***********************
 and this is server console response (2ndhalf only after VLC stream query) when I switch input from 2 (component) to SVideo or Composite (both are unused). My Slingbox is M1. Hope this helps come up with some conclusion. 


Slinginfo  "350/500/M1" 14i7p88p0g194gy 12 30 192.168.0.241 5301 3072
PLM1UPC Starting Stream for  192.168.0.220:59175
Connecting... ('192.168.0.241', 5301) Control
PLM1UPC Selecting Video Source 0
Keycodes= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 21, 24, 25, 26, 27, 28, 29, 32, 33, 34, 35, 37, 38, 39, 40, 41, 42, 43, 44, 46, 47, 62, 63, 67, 69, 72, 73, 74, 75, 76, 77]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 32
Connecting... ('192.168.0.241', 5301) Stream
PLM1UPC Stream started at 09/30/2022, 17:36:20  744 0
09/30/2022, 17:36:23  PLM1UPC Stream Terminated for  192.168.0.220:59175 [WinError 10053] An established connection was aborted by the software in your host machine
PLM1UPC Shutting down connections
Streamer:  PLM1UPC Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Reply by GerryDazoo** [reply]
The current code doesn't handle the case where the slingbox returns an error when setting the video source. The M1 I use for testing works correctly with VideoSource=2. Are you sure It's properly configured?
I'm afraid I don't know what to tell you. Is your M1 still dropping off the network?
Just for fun, start a video with the Sling software. Then comment out the VideoSource line  ;VideoSource=2
And we'll see what happens.

**Reply by eskwadrat** [reply]
Thanks for taking the time to troubleshoot it, I really appreciate it. 

Yes, my M1 is configured properly and works for almost 3 yrs now with Sling software. Better than any other Slingbox I used before (Pro, Solo). The Component input is used (YPbPr) and it works with Sling software showing crystal clean picture in 720p, on the VLC I get nothing. The M1 is no longer dropping off the network. I did start the Sling software and started to stream via it. Then I ran your server with commented ;VideoSource=2, as soon as I ran VLC, Sling software shows the message:

"The box is used by someone else, do you want to disconnect them?"
[Yes] [No]

When I answer yes, the Sling software returns to stream playback,while server console remains as before. Then it adds few lines below.
Note: I had a thought, perhaps Slingbox video stream configuration is different from one defined in the server .ini file (res is the same but framerate, etc might not be what Slingbox is streaming at), hence the issue. 

Version : 3.07 Running on Windows-10-10.0.19042-SP0 pid= 3176
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running:  10 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('192.168.0.241', 5301)
('192.168.0.241', 5301) OK
Using slingbox at  ('192.168.0.241', 5301)
Streamer:  PLM1UPC Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses (0.0.0.0)
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://127.0.0.1:8081
 * Running on http://192.168.0.220:8081 (Press CTRL+C to quit)
09/30/2022, 19:04:05   Streaming connection from ('192.168.0.220', 59448)

Slinginfo  "350/500/M1" 14i7p88p0g194gy 12 30 192.168.0.241 5301 3072
PLM1UPC Starting Stream for  192.168.0.220:59448
Connecting... ('192.168.0.241', 5301) Control
cmd: 0x7e err: 0x13 176
PLM1UPC Box in use! Kicking off other user.
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 32
Connecting... ('192.168.0.241', 5301) Stream
PLM1UPC Stream started at 09/30/2022, 19:04:10  744 0
09/30/2022, 19:04:11  PLM1UPC Stream Terminated for  192.168.0.220:59448 [WinError 10053] An established connection was aborted by the software in your host machine
PLM1UPC Shutting down connections
Streamer:  PLM1UPC Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Reply by JoeM2016** [reply]
This is my first attempt at using this forum.  Any suggestions for getting past this point.  I think I followed all of the above suggestions for setting up the config file and can access the SB500 with the SB Desktop App. I am just trying to start by using it on my LAN before I extend it to the outside. 
![SB Server](https://user-images.githubusercontent.com/114776897/193355467-288a25cf-531a-47a2-b005-8259e64d30bb.jpg)

**Comment by eskwadrat** [comment]
GerryDazoo: guess what! I just updated VLC and the VLC WORKS! Yupee!
Not getting the remote part... so happy.

here is what console says:

Slinginfo  "350/500/M1" 14i7p88p0g194gy 12 30 192.168.0.241 5301 3072
PLM1UPC Starting Stream for  192.168.0.220:59574
Connecting... ('192.168.0.241', 5301) Control
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 32
Connecting... ('192.168.0.241', 5301) Stream
PLM1UPC Stream started at 09/30/2022, 19:18:27  744 0
........09/30/2022, 19:19:58  PLM1UPC 1 Clients.192.168.0.220:59574
..

**Reply by eskwadrat** [reply]
Premature conclusion. It works only when I run Sling software first and then run VLC to kick the Sling software stream. 
But I found something. Whenever I have VideoSource=2 parameter active, the VLC won't play a thing and my Slingbox goes offline after closing server. I need to power cycle it to get it back online. 

But when I comment ;VideoSource, then VLC plays just fine. I know from your comments in the .ini file that it plays last used input, so it looks like in my M1 it is not the 2. Not sure what is it cause I tried 0 and 1 and it did not work either. Will try 3 instead.

**Reply by sp222222** [reply]
I left my videosource commented out on most of mine as it was also causing issues.(I have six slingboxes currently working) once you get it working using the normal sling app go in and rem (semicolon) that videosource line out and the slinger app should should video fine.

**Comment by eskwadrat** [comment]
I confirm, in my M1 the component input is 3 and it works. 

VideoSource=3

**Reply by GerryDazoo** [reply]
That's  interesting. Maybe we have different M1 models. So can you enumerate all your video inputs for your box?

**Reply by eskwadrat** [reply]
I have 2nd M1 box in another house overseas, will try the same on it and report. 

Now let's try solving remote part... YOU'RE MY HERO.

**Comment by eskwadrat** [comment]
Remote works also, just need to remap some buttons to fit my UPC Horizon (PACE DMC7002KLG) set top box. Not sure if what's in the remote.txt is kinda universal command set? Or is it something that is different for each source make/model?

**Comment by eskwadrat** [comment]
I take VideoSource=3 back. I had it set to 3 but forgot to remove comment. After I noticed my error, I removed the comment sign leaving 3. It does not work with this setting either. The only time the server works when I run it with .ini lacking VideoSource setting at all (last used video source is active). 

Btw, I power cycled the M1 and it seem to remember last video source used so this VideoSource issue is no longer threat of loosing the stream next time I use it. I just use .ini without it defined. And it works everytime. 

I hope I can find good pointer on how to set port forwarding to make it work remotely via my dyndns.

**Comment by eskwadrat** [comment]
OK, port forwarding works, I can watch it and control now from anywhere. 

Note: the most tough part for me in past couple years since SlingMedia announced servers shutdown... was not the video but to get my stb remote control working. The PACE DMC7002 stb I have (UPS Horizon) has Bluetooth remote by standard so you can't use it to teach 3rd party cloud based remote solutions like Broadlink. Still, this PACE has also IR receiver which was the only way I could use it for many years in the past with the Slingboxes which use IR remote control as well. 

Not knowing the IR command set for my STB I even tried to teach universal remotes by generating IR commands from the Slingbox IR emitter, for some reason this method does not work, the universal remotes like Broadlink or SmatIR can't learn it right from SlngBox so when I try checking learning result, it fails. I think the reason might be Slngbox IR emitter sending only single control packet while to teach universal remote you need more than one to get it right. 
I was trying to find Ethernet based method to control the STB but the only reference I found was for different older STB made by UPC which was hacked y rogro82 (github rep "WiFi Remote for Horizon"). But I am not java or python programmer and not familiar with modern software layers programming so I was stuck. 

The remote Gerry uses is my first successful solution to change the channels when streaming video via another non-Sling streamers, like J-TECH Digital JTECH-ENCH5 (which uses H265 codes offering 4K streaming using same bandwidth that Slingbox HD was using). I have two of these streamers and they work amazing with 4K sources I have scattered overseas.  

Ideally, I would expect and even pay developer to develop simple Android app for "Remote IR remote" what would allow me to control my STBs using old Slingboxes  (I have 3 which collect dust) or perhaps even better - non Slingbox based remotes. With video streamed via modern H265 streamer like the one I use above. 

Just a food for thought.

**Reply by flipper2222222** [reply]
I use a harmony hub with a Pace box, about 5 secs lag on the remote change.

**Reply by eskwadrat** [reply]
Crazy expensive. I managed to make $29 Broadlink Mini4 working with my Horizon UPC stb.
Now I do not even need Slingbox anymore with my newer H265 streamer.

**Comment by GerryDazoo** [comment]
Somebody type ctrl/c to kill the program
Based on the logs you posted everything look fine.
Send a streaming request to the server.
http://192.168.1.108:8080/slingbox 

 

>

**Reply by JoeM2016** [reply]
Thanks, it started working when I used the server's address in the client (VLC app on an iPad).  I misinterpreted an earlier replay for previous query.  I hit the control C when it appeared it was not working.
